src/Uno.UWP/Globalization/NumberFormatting/NumeralSystemTranslator.cs

```diff
@@ -0,0 +1,167 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Windows.Globalization.NumberFormatting
+{
+	public partial class NumeralSystemTranslator
+	{
+		private static Dictionary<string, char[]> s_numeralSystemToDigitsDictionary =
+			NumeralSystemTranslatorHelper.CreateNumeralSystemToDigitsDictionary();
+		private string numeralSystem;
+
+		public string NumeralSystem
+		{
+			get => numeralSystem;
+			set
+			{
+				if (!s_numeralSystemToDigitsDictionary.ContainsKey(value))
+				{
+					throw new ArgumentException("The parameter is incorrect.");
+				}
+
+				numeralSystem = value;
+			}
+		}
+
+		public IReadOnlyList<string> Languages { get; }
+
+		public string ResolvedLanguage { get; }
+
+		public NumeralSystemTranslator(IEnumerable<string> languages)
+		{
+			ValidateLanguages(languages);
+
+			Languages = languages.ToList();
+			ResolvedLanguage = NumeralSystemTranslatorHelper.GetResolvedLanguage(Languages[0]);
+			NumeralSystem = NumeralSystemTranslatorHelper.GetNumeralSystem(Languages[0]);
+		}
+		public NumeralSystemTranslator() : this(new string[] { "en-US" })
+		{
+		}
+
+		private void ValidateLanguages(IEnumerable<string> languages)
+		{
+			if (languages is null)
+			{
+				throw new NullReferenceException("Invalid pointer");
+			}
+
+			if (!languages.Any())
+			{
+				throw new ArgumentException("The parameter is incorrect.");
+			}
+
+			foreach (var language in languages)
+			{
+				NumeralSystemTranslatorHelper.GetNumeralSystem(language);
+			}
+		}
+
+		public string TranslateNumerals(string value)
+		{
+			if (NumeralSystem == "Arab" ||
```

Comment:
Please perform the comparison using `.Equals(..., CultureInfo.InvariantCultureIgnoreCase)`